src/AudioIO.cpp

```diff
@@ -3812,7 +3828,7 @@ int audacityAudioCallback(const void *inputBuffer, void *outputBuffer,
 {
    auto gAudioIO = AudioIO::Get();
    return gAudioIO->AudioCallback(
-      inputBuffer, outputBuffer, framesPerBuffer,
+      (constSamplePtr)inputBuffer, (float*)outputBuffer, framesPerBuffer,
```

Comment:
static_casts maybe?

Reply:
Yes.

I don't like the type aliases samplePtr and constSamplePtr, for char* and const char*, which often travel together with a sampleFormat telling you how to interpret the data.  Better to put the pointer and the format in a class.  But doing that cleanup, which will eliminate still more casts, is for later.